DEVELOPMENT.md

```diff
@@ -0,0 +1,71 @@
+# Development Guide
+
+This document describes the high level workflow used when working on a WordPress Plugin or Theme.
+
+You're free to use your preferred IDE and Git client. We recommend PHPStorm or Visual Studio Code, and GitHub CLI.
+
+## Prerequisites
+
+If you haven't yet set up your local development environment with a WordPress Plugin repository installed, refer to the [Setup Guide](SETUP.md).
+
+## Create a Branch
+
+In your Git client / command line, create a new branch:
+- If this is for a new feature that does not have a GitHub Issue number, enter a short descriptive name for the branch, relative to what you're working on
+- If this is for a feature/bug that has a GitHub Issue number, enter issue-XXX, replacing XXX with the GitHub issue number
```

Comment:
Maybe we can update the L15; as I know from the past, Andrei and @cristian-ungureanu don't suggest creating a branch via issue id.